probe-rs/src/architecture/arm/core/armv6m.rs

```diff
@@ -636,15 +636,15 @@ impl<'probe> CoreInterface for Armv6m<'probe> {
 
     fn step(&mut self) -> Result<CoreInformation, Error> {
         // First check if we stopped on a breakpoint, because this requires special handling before we can continue.
-        let pc_before_step = self.read_core_reg(self.program_counter().into())?;
-        let was_breakpoint = if matches!(
+        let (was_breakpoint, pc_before_step) = if matches!(
             self.state.current_state,
             CoreStatus::Halted(HaltReason::Breakpoint(_))
         ) {
+            let pc_before_step = self.read_core_reg(self.program_counter().into())?;
             self.enable_breakpoints(false)?;
-            true
+            (true, pc_before_step)
```

Comment:
`Some(pc_before_step)`

Reply:
wouldn't this require `unwrap()` or `match` later when using `pc_before_step`?

Reply:
ok thanks. will change it :)